packages/middleware-signing/src/middleware.ts

```diff
@@ -20,10 +27,24 @@ export function awsAuthMiddleware<Input extends object, Output extends object>(
       args: FinalizeHandlerArguments<Input>
     ): Promise<FinalizeHandlerOutput<Output>> {
       if (!HttpRequest.isInstance(args.request)) return next(args);
-      return next({
+      const { response } = await next({
         ...args,
-        request: await options.signer.sign(args.request)
+        request: await options.signer.sign(args.request, {
+          signingDate: new Date(Date.now() + options.systemClockOffset)
+        })
       });
+
+      const { headers } = response as any;
+      const dateHeader = headers && (headers.date || headers.Date);
+      if (dateHeader) {
+        const serverTime = Date.parse(dateHeader);
+        if (isClockSkewed(serverTime, options.systemClockOffset)) {
+          options.systemClockOffset = serverTime - Date.now();
+        }
+      }
+
+      // @ts-ignore
+      return { response };
```

Comment:
Other members in the next handler's output might missing here, like [`output`](https://github.com/aws/aws-sdk-js-v3/blob/4d3a5b00db1d632c5547b19ae68923868007decd/packages/types/src/middleware.ts#L52-L63).  It safer to return all other things from `next`

Reply:
Thanks for catching this!
Resolved in https://github.com/aws/aws-sdk-js-v3/pull/459/commits/7cd7d2487dafa00600c4e7d10e4caa7e98b015e8